content/en/docs/setup/production-environment/tools/kubeadm/dual-stack-support.md

```diff
@@ -0,0 +1,144 @@
+
+<!-- body -->
+
+## Enable dual-stack via kubeadm
+
+Install kubeadm following the steps from the [Installing Kubeadm](/docs/setup/production-environment/tools/kubeadm/install-kubeadm/) documentation.
+
+Make sure that nodes allow IPv6 forwarding, if not, run `sudo sysctl -w net.ipv6.conf.all.forwarding=1` on every node in the cluster.
+
+{{< note >}}
+`kubeadm upgrade` will change `IPv6DualStack` to true by default if the feature gate is not set in an old cluster. However, cluster CIDR and service CIDR modifications are not supported.
+{{< /note >}}
+
+### Create a dual-stack cluster
+
+To create a dual-stack cluster with `kubeadm init` you can pass the following flags:
+
+```shell
+kubeadm init --pod-network-cidr=10.244.0.0/16,fd00:10:244::/56 --service-cidr=10.96.0.0/16,fd00:10:96::/112
+```
+
+To make things clearer, here is an example kubeadm [configuration file](https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2) `kubeadm-config.yaml` for the primary dual-stack control plane node.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: ClusterConfiguration
+featureGates:
+  IPv6DualStack: true
+networking:
+  podSubnet: 10.244.0.0/16,fd00:10:244::/56
+  serviceSubnet: 10.96.0.0/16,fd00:10:96::/112
+---
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: InitConfiguration
+localAPIEndpoint:
+  advertiseAddress: "10.100.0.1"
+  bindPort: 6443
+nodeRegistration:
+  kubeletExtraArgs:
+    fail-swap-on: "false"
+    node-ip: 10.100.0.2,fd00:1:2:3::2
+```
+
+`advertiseAddress` in InitConfiguration specifies the IP address the API Server will advertise it's listening on. It equals to the `--apiserver-advertise-address` flag of `kubeadm init`.
+
+Run kubeadm to initiate the dual-stack control plane node.
+
+```shell
+kubeadm init --config=kubeadm-config.yaml
+```
+
+Currently, the kube-controller-manager flags `--node-cidr-mask-size-ipv4|--node-cidr-mask-size-ipv6` are being left with default values. See [enable IPv4/IPv6 dual stack](/docs/concepts/services-networking/dual-stack#enable-ipv4ipv6-dual-stack).
+
+{{< note >}}
+Currently the `--apiserver-advertise-address` flag does not support dual-stack.
+{{< /note >}}
+
+### Join a node to dual-stack cluster
+
+Before joining a node, make sure that the node has IPv6 routable network interface and allows IPv6 forwarding.
+
+Here is an example kubeadm [configuration file](https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2) `kubeadm-config.yaml` for joining a worker node to the cluster.
+
+```yaml
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: JoinConfiguration
+discovery:
+  bootstrapToken:
+    apiServerEndpoint: 10.100.0.1:6443
+    token: 0c0z4p.dnafh6vnmouus569
+    caCertHashes: ["sha256:fcb3e956a6880c05fc9d09714424b827f57a6fdc8afc44497180905946527adf"]
+nodeRegistration:
+  kubeletExtraArgs:
+    fail-swap-on: "false"
```

Comment:
Why set `fail-swap-on`?

Reply:
yep, this can be omitted.
we can also keep the config minimal without non-relevant values like the kubelet extra args and CA-hash / tokens.

Reply:
removed.